packages/did-ethr-resolver/src/implementations/operator.ts

```diff
@@ -74,14 +78,15 @@ export class Operator extends Resolver implements IOperator {
     return this.address || this._signer.getAddress();
   }
 
-  private async getPublicKey(): Promise<string> {
+  public async getPublicKey(): Promise<string> {
     if (this._keys.publicKey) {
       return this._keys.publicKey;
     }
-    const hash = await ethers.utils.keccak256(await this.getAddress());
-    const sig = await this._signer.signMessage(ethers.utils.arrayify(hash));
+    const hash = keccak256(await this.getAddress());
+    const sig = await this._signer.signMessage(arrayify(hash));
+    const digest = arrayify(hashMessage(arrayify(hash)));
     // eslint-disable-next-line no-return-assign
-    return this._keys.publicKey = ethers.utils.recoverPublicKey(hash, sig);
+    return this._keys.publicKey = `02${recoverPublicKey(digest, sig).slice(4, 68)}`;
```

Comment:
same here. 02?

Reply:
Prefix used to differentiate between compressed and uncompressed key